src/app/screens/Login/index.js

```diff
@@ -23,25 +25,41 @@ class LoginContainer extends Component {
         this.setState({ invalidPassword: password.length < 8 });
     };
 
-    loginOnPressHandler = () => {
-        const { navigation } = this.props;
-        navigation.navigate(Home);
+    loginOnPressHandler = async () => {
+        const { navigation, signIn } = this.props;
+        const { email, password } = this.state;
+        await signIn(email, password);
+        const { signInErrorMessage } = this.props;
```

Comment:
add this to previous props

Reply:
I must have to get the `signInErrorMessage` prop value after await the asynchronous call to signIn  in order to get the updated value. Otherwise a invalid password that fail in signIn function will not be detected, allowing access to library with all king of incorrect passwords.

Reply:
then you should manage this action inside redux instead in the component